### DISCUSSION TITLE
using nbconvert to create a python script from a notebook

### DISCUSSION THREAD

**Discussion by fearon10** [ARM-Development/comble-mip#33] [General]
Hi There,

I was wondering if anyone has attempted to use nbconvert to convert a jupyter notebook to a python script? I would like to do this as script format would be easier to use in my current working environment. However, it my attempts to use nbconvert I receive the following validation error. Just wondering if others might have a solution to this issue.

 jupyter nbconvert --to python create_comble_forcing_v2.3.ipynb

 raise ValidationError(msg) from None
jsonschema.exceptions.ValidationError: Notebook could not be converted from version 1 to version 2 because it's missing a key: cells

Thanks,
Matt

**Comment by twjuliano** [comment]
Hi Matt,

I was able to use the same command and produce a .py file from the .ipynb file on ARM’s Workbench JupyterHub:

![image](https://github.com/ARM-Development/comble-mip/assets/46979614/5847305c-bf65-4c02-bdf2-a62222e6ce98)
 
I wonder if the issue is related to your version of jupyter? Here are my versions:

![image](https://github.com/ARM-Development/comble-mip/assets/46979614/af5c07ba-b1d2-4e0a-91db-7b1646180f3e)

As an aside, we encourage participants to apply for elevated access to the ARM Workbench, which you can do [here](https://arm-development.github.io/comble-mip/notebooks/howto/workbench_instructions.html).

I hope this information helps -- please let us know if you're able to resolve the issue.

Tim

**Reply by fearon10** [reply]
Hi Tim,

Apologies for the really long delay in responding. I do sincerely appreciate your assistance. Too many projects have kept me from testing your suggestion of a new specific jupyter version. However, I finally did try this, i.e., install jupyterlab 3.6.1 and notebook 6.5.3. I still receive the same error; however, it may be that I don't have all the dependent package versions exactly the same as you. Some of the dependencies are version or two ahead, see below. I wonder if I need a different version of python. Could you please share which version of python you're using? Otherwise, I'm not sure. Some of the dependencies are installed automatically. 

Thanks, Matt


My install steps:
conda create -n pycomble python=3.10
conda activate pycomble
pip install ipython==8.11.0
pip install ipykernel==6.21.3
pip install jupyterlab==3.6.1
pip install notebook==6.5.3
pip install ipywidgets==8.0.4


jupyter --version
Selected Jupyter core packages...
IPython          : 8.11.0
ipykernel        : 6.21.3
ipywidgets       : 8.0.4
jupyter_client   : 7.4.9
jupyter_core     : 5.7.1
jupyter_server   : 2.12.5
jupyterlab       : 3.6.1
nbclient         : 0.9.0
nbconvert        : 7.14.2
nbformat         : 5.9.2
notebook         : 6.5.3
qtconsole        : not installed
traitlets        : 5.14.1

**Reply by fearon10** [reply]
Hi Tim,

Since there were a few of the discrepancies in the version on the dependencies, I went back through with pip and installed the exact version matching yours. Still no dice, same error. 

jupyter nbconvert --to python create_comble_forcing_v2.3.ipynb
    raise ValidationError(
jsonschema.exceptions.ValidationError: Notebook could not be converted from version 1 to version 2 because it's missing a key: cells

versions:
Python 3.8.18

jupyter --version
Selected Jupyter core packages...
IPython          : 8.11.0
ipykernel        : 6.21.3
ipywidgets       : 8.0.4
jupyter_client   : 8.0.3
jupyter_core     : 5.3.0
jupyter_server   : 2.5.0
jupyterlab       : 3.6.1
nbclient         : 0.7.2
nbconvert        : 7.2.9
nbformat         : 5.7.3
notebook         : 6.5.3
qtconsole        : not installed
traitlets        : 5.9.0
